### PR TITLE
closes STORM-1545 and STORM-1552

### PR DESCRIPTION
**Pull request by mproch**
Use ConfigUtils.workerArtifactsRoot to determine correct path

### CODE REVIEW

File: storm-core/src/jvm/org/apache/storm/metric/FileBasedEventLogger.java [hunk 1]
```diff
@@ -79,26 +80,18 @@ public void run() {
...
+
+        String workersArtifactRoot = ConfigUtils.workerArtifactsRoot(stormConf, stormId, port);
 
         /*
          * Include the topology name & worker port in the file name so that
          * multiple event loggers can log independently.
          */
-        Path path = Paths.get(logDir, "workers-artifacts", stormId, Integer.toString(port), "events.log");
+        Path path = Paths.get(workersArtifactRoot, "events.log");
         if (!path.isAbsolute()) {
```
Comment (arunmahadevan) [inline storm-core/src/jvm/org/apache/storm/metric/FileBasedEventLogger.java]: [ConfigUtils](https://github.com/apache/storm/blob/master/storm-core/src/jvm/org/apache/storm/utils/ConfigUtils.java#L407) appears to take care of the relative path, so this check could be removed.

**Comment by arunmahadevan** [comment]
Can you update the PR title ?

Theres a PR https://github.com/apache/storm/pull/1109 for the same issue but it does not make use of the ConfigUtils since it had to be merged to 1.x as well. Since this PR is a better fix for 2.0, I will close https://github.com/apache/storm/pull/1109 and raise that against 1.x-branch.

+1 to merge against master (not 1.x-branch) once review comment is addressed.

**Comment by mproch** [comment]
I removed relative path handling and updated the title - hope this is what you meant?

**Comment by arunmahadevan** [comment]
When the unit tests are run it creates log files under /logs. It should ideally be under system tmp directory. 

Maybe unrelated, but when an example topology is submitted worker crashes with the below error,

`2016-02-22 10:56:51.085 o.a.s.d.worker [ERROR] Error on initialization of server mk-worker
java.lang.ClassCastException: java.lang.String cannot be cast to java.io.File
  at org.apache.storm.daemon.worker$fn__6265$exec_fn__1730__auto____6266.invoke(worker.clj:622) ~[storm-core-2.0.0-SNAPSHOT.jar:2.0.0-SNAPSHOT]
  at clojure.lang.AFn.applyToHelper(AFn.java:178) ~[clojure-1.7.0.jar:?]
  at clojure.lang.AFn.applyTo(AFn.java:144) ~[clojure-1.7.0.jar:?]
  at clojure.core$apply.invoke(core.clj:630) ~[clojure-1.7.0.jar:?]
  at org.apache.storm.daemon.worker$fn__6265$mk_worker__6362.doInvoke(worker.clj:613) [storm-core-2.0.0-SNAPSHOT.jar:2.0.0-SNAPSHOT]
  at clojure.lang.RestFn.invoke(RestFn.java:512) [clojure-1.7.0.jar:?]
  at org.apache.storm.daemon.worker$_main.invoke(worker.clj:810) [storm-core-2.0.0-SNAPSHOT.jar:2.0.0-SNAPSHOT]
  at clojure.lang.AFn.applyToHelper(AFn.java:165) [clojure-1.7.0.jar:?]
  at clojure.lang.AFn.applyTo(AFn.java:144) [clojure-1.7.0.jar:?]
  at org.apache.storm.daemon.worker.main(Unknown Source) [storm-core-2.0.0-SNAPSHOT.jar:2.0.0-SNAPSHOT]`

**Comment by satishd** [comment]
https://issues.apache.org/jira/browse/STORM-1566 is raised for the above issue.

**Comment by abhishekagarwal87** [comment]
Thanks @mproch Looks good to me (NB) for merge with master.

**Comment by HeartSaVioR** [comment]
+1

**Comment by HeartSaVioR** [comment]
@mproch 
Thanks for the work! I merged this into master.

I'd like to also set assignee from JIRA issue to you but I couldn't find your name in Apache JIRA.
If you have your account in JIRA, please do the followings,

- set assignee to yourself at https://issues.apache.org/jira/browse/STORM-1545 
- resolve issue as 'resolved'
- set fix version/s to 2.0.0

Please let me know if you're not interested to do the things. I'll just assign assignee as reporter and resolve issue.

**Comment by mproch** [comment]
@HeartSaVioR 
thanks, I resolved jira issue.

**Pull request merged**